RadarSDK/RadarTripOptions.m

```diff
@@ -0,0 +1,82 @@
+//
+//  RadarTripOptions.m
+//  RadarSDK
+//
+//  Copyright © 2020 Radar Labs, Inc. All rights reserved.
+//
+
+#import "RadarTripOptions.h"
+
+@implementation RadarTripOptions
+
+NSString *const kExternalId = @"externalId";
+NSString *const kMetadata = @"metadata";
+NSString *const kDestinationGeofenceTag = @"destinationGeofenceTag";
+NSString *const kDestinationGeofenceExternalId = @"destinationGeofenceExternalId";
+NSString *const kMode = @"mode";
+
+- (instancetype)initWithExternalId:(NSString *_Nonnull)externalId {
+    self = [super init];
+    if (self) {
+        _externalId = externalId;
+        _mode = RadarRouteModeCar;
+    }
+    return self;
+}
+
++ (RadarTripOptions *)tripOptionsFromDictionary:(NSDictionary *)dict {
+    if (!dict) {
+        return nil;
+    }
+
+    RadarTripOptions *options = [[RadarTripOptions alloc] initWithExternalId:dict[kExternalId]];
+    options.metadata = dict[kMetadata];
+    options.destinationGeofenceTag = dict[kDestinationGeofenceTag];
+    options.destinationGeofenceExternalId = dict[kDestinationGeofenceExternalId];
+    NSString *modeStr = dict[kMode];
+    if ([modeStr isEqualToString:@"foot"]) {
+        options.mode = RadarRouteModeFoot;
+    } else if ([modeStr isEqualToString:@"bike"]) {
+        options.mode = RadarRouteModeBike;
+    } else {
+        options.mode = RadarRouteModeCar;
+    }
+    return options;
```

Comment:
nil checking may be needed for @"mode" and "externalId"

Reply:
I'll add `nil` checks for `externalId` and `modeStr` to be extra safe